installer/x86_64/install.sh

```diff
@@ -583,19 +583,20 @@ EOF
     $onie_root_dir/tools/bin/onie-boot-mode -q -o install
 fi
 
-# Add a menu entry for the DEMO OS
+# Add a menu entry for the SONiC OS
 # Note: assume that apparmor is supported in the kernel
 demo_grub_entry="$demo_volume_revision_label"
 if [ "$install_env" = "sonic" ]; then
     old_sonic_menuentry=$(cat /host/grub/grub.cfg | sed "/$running_sonic_revision/,/}/!d")
-    demo_dev=$(echo $old_sonic_menuentry | sed -e "s/.*root\=\(.*\)rw.*/\1/")
+    grub_cfg_root=$(echo $old_sonic_menuentry | sed -e "s/.*root\=\(.*\)rw.*/\1/")
     onie_menuentry=$(cat /host/grub/grub.cfg | sed "/menuentry ONIE/,/}/!d")
-fi
-
-if [ "$install_env" = "build" ]; then
+elif [ "$install_env" = "build" ]; then
     grub_cfg_root=%%SONIC_ROOT%%
 else
-    grub_cfg_root=$demo_dev
+    grub_cfg_root=$(blkid "$demo_dev" | grep -oE "[[:blank:]]UUID=\".*\"" | sed s/\"//g | sed s/\ //g)
```

Comment:
>blkid "$demo_dev" | grep -oE "[[:blank:]]UUID=\".*\"" | sed s/\"//g | sed s/\ //g [](start = 20, length = 81)

```suggestion
    grub_cfg_root=$(blkid "$demo_dev" | sed 's/.*UUID=\"\([^"]*\)\".*/\1/g')
```

Reply:
the problem of the suggestion is that if there is UUID entry in the output, you will the get whole string. test following.

```
/dev/vda3: LABEL="SONiC-OS" PARTUUID="abc" DUUID="0530e631-21fa-4707-bbf0-6100631556e3" PARTUUID="xxx"
/dev/vda3: LABEL="SONiC-OS" PARTUUID="abc"
/dev/vda3: LABEL="SONiC-OS"
```

Reply:
I see. I refine the suggestion. You may commit it if you feel it is good.

---
In reply to: [410710790](https://github.com/Azure/sonic-buildimage/pull/4443#discussion_r410710790) [](ancestors = 410710790)

Reply:
thanks